qmi/instruments/picotech/picoscope3403.py

```diff
@@ -106,43 +128,59 @@ def acquire_by_trigger_and_get_data(
+        the number of samples before the current moment of `run_block` call.
+        The data is then sent back as real voltages.
 
         Parameters:
-            channels: List of used channels [0...NUM_CHANNELS-1] for readout.
-            max_val: Maximum expected values for ALL channels [0, 1, ... NUM_CHANNELS-1] to set an appropriate
-                input range.
-            couplings: Coupling of channels for readout for ALL channels [0, 1, ... NUM_CHANNELS-1].
-                ChannelCoupling.DC or ChannelCoupling.AC.
-            trigger_channel: Channel to set the trigger on. If it is 'None', triggers will be disabled on all channels.
-            trigger_level: Threshold in Volts for the trigger.
-            time_base: Data acquirement time resolution in effective time base: `(time_base+1) * 12.5 ns`.
-            time_span: Time span of data acquiring in time base as above. Thus not in ns.
+            channels:          List of used channels [0...NUM_CHANNELS-1] for readout.
+            max_val:           Maximum expected values for ALL channels [0, 1, ... NUM_CHANNELS-1] to set an
+                               appropriate input range.
+            couplings:         Coupling of channels for readout for ALL channels [0, 1, ... NUM_CHANNELS-1].
+                               ChannelCoupling.DC or ChannelCoupling.AC.
+            trigger_channel:   Channel to set the trigger on. If it is 'None', triggers will be disabled on
+                               all channels.
+            trigger_level:     Threshold in Volts for the trigger.
+            sampling_interval: Sampling interval in nanoseconds. Must be 1, 2, 4 or a multiple of 8 ns.
+            time_span:         Time span of data acquiring in nanoseconds.
 
         Returns:
-            Tuple `(times,voltages)`.
-            `times` is a 1D Numpy array containing the measured time in us.
-            `voltages` is a 1D Numpy array containing the measured voltages in V.
+            (times, voltages): `times` is a 1D Numpy array containing the sampling times in ns.
+                               `voltages` is a 1D Numpy array containing the measured voltages in V.
         """
+        warnings.warn(
+            f"{self.acquire_by_trigger_and_get_data.__name__} will be deprecated in the future as it is not a base "
+            "functionality of the driver. Please implement it as a custom method in your code.")
+
+        if min(channels) < 0 or max(channels) >= self.NUM_CHANNELS:
+            raise ValueError("Invalid channel index")
+
         if trigger_channel is not None:
             if (trigger_channel < 0) or (trigger_channel >= self.NUM_CHANNELS) or trigger_channel not in channels:
                 raise ValueError("Invalid trigger channel index")
 
-        max_volts = list()
+        time_base = self._get_time_base(sampling_interval)
+        if time_base < 2 and len(channels) > 2 ** time_base:
```

Comment:
if `time_base` is 2 and there are 5 channels then this condition is not triggered and it should be triggered right?

Reply:
Well it is irrelevant as there are only up to 4 channels.